### PR TITLE
Cromwell should be able to take a hint.

### PR DESCRIPTION
**Pull request by mcovarr**
Allows parsing hints and requirements. ~~The requirement parsing is more liberal than it should be, I'm still investigating that.~~ No I'm just a dummy and was feeding my Cromwell the wrong copy of the `three_step.cwl` file.

### CODE REVIEW

File: cwl/src/test/resources/three_step.cwl [hunk 1]
```diff
@@ -1,6 +1,15 @@
 cwlVersion: v1.0
 class: Workflow
 id: three_step
+hints:
+  # hints can be anything (CwlAny)
+  - fee: fi
+    fo: fum
+  - eenie: meenie
+    minie: mo
+  - 42
```
Comment (cjllanwarne) [inline cwl/src/test/resources/three_step.cwl]: ToL: I don't know whether the `three_step.cwl` is the best place to be testing the limits of `CwlAny`'s parsing

Comment (mcovarr) [reply cwl/src/test/resources/three_step.cwl]: It's not, this is a quick hack. 🙂

**Comment by mcovarr** [comment]
Lovely, this parses and runs when actually submitted to Cromwell but the unit tests won't even parse it 😡

**Comment by mcovarr** [comment]
Ok talked with Dan and there are a couple of problems here: 
We have two copies of three_step.cwl, one of which is used by unit tests and the other is used by Centaur. The unit tests currently don't SALAD input CWL so their input needs to be formatted just so. Real Cromwell does SALAD input CWL so it's a lot more flexible. Dan is working on a separate PR so the unit tests run SALAD.

File: cwl/src/main/scala/cwl/Workflow.scala [hunk 2]
```diff
@@ -25,7 +25,9 @@ case class Workflow private(
                      id: String,
                      inputs: Array[InputParameter],
                      outputs: Array[WorkflowOutputParameter],
-                     steps: Array[WorkflowStep]) {
+                     steps: Array[WorkflowStep],
+                     requirements: Option[Array[Requirement]],
+                     hints: Option[Array[CwlAny]]) {
```
Comment (cjllanwarne) [inline cwl/src/main/scala/cwl/Workflow.scala]: These must also be added to `ExpressionTool`

Comment (mcovarr) [reply cwl/src/main/scala/cwl/Workflow.scala]: Strangely enough, they are already there (I did not add them). 😮

Comment (cjllanwarne) [reply cwl/src/main/scala/cwl/Workflow.scala]: Huh, I found my CNil was due to an errant `ExpressionTool`, I guess it wasn't the requirement that it wasn't liking after all 😨

Comment (mcovarr) [reply cwl/src/main/scala/cwl/Workflow.scala]: Unit test or Centaur? Unit tests currently are not SALADing so you have to format requirements just so.

**Pull request merged**